app/controllers/posts_controller.rb

```diff
@@ -1,8 +1,9 @@
 class PostsController < ApplicationController
   def index
     # For displaying all posts by a given user
-    @user = User.find(params[:user_id])
+    @user = User.includes(posts: :comments).find(params[:user_id])
     @posts = @user.posts
+    @comments = @posts.map { |post| post.comments.limit(5) }
   end
```

Comment:
- [ ] [REQUIRED] You have done a great job so far, however, this `index` method doesn't completely solve the N+1 query issue. While you are eager loading posts and comments associated with each post using includes, the subsequent `@comments = @posts.map { |post| post.comments.limit(5) }`  introduces a potential N+1 problem again.

### To resolve the N + 1 problem completely
``` ruby
def index
  @user = User.includes(posts: [:comments]).find(params[:user_id])
  @posts = @user.posts.includes(:comments)
end
```

- Now, both the user's posts and associated comments are loaded eagerly in a more optimized way. Accessing @posts in your view should not result in additional queries for comments because they have already been loaded.